deepchem/models/tensorgraph/progressive_multitask.py

```diff
@@ -0,0 +1,284 @@
+
+    n_layers = len(layer_sizes)
+    if not isinstance(weight_init_stddevs, collections.Sequence):
+      self.weight_init_stddevs = [weight_init_stddevs] * n_layers
+    if not isinstance(alpha_init_stddevs, collections.Sequence):
+      self.alpha_init_stddevs = [alpha_init_stddevs] * n_layers
+    if not isinstance(bias_init_consts, collections.Sequence):
+      self.bias_init_consts = [bias_init_consts] * n_layers
+    if not isinstance(dropouts, collections.Sequence):
+      self.dropouts = [dropouts] * n_layers
+    if not isinstance(activation_fns, collections.Sequence):
+      self.activation_fns = [activation_fns] * n_layers
+
+    # Add the input features.
+    self.mol_features = Feature(shape=(None, n_features))
+
+    all_layers = {}
+    outputs = []
+    for task in range(self.n_tasks):
+      task_layers = []
+      for i in range(n_layers):
+        if i == 0:
+          prev_layer = self.mol_features
+        else:
+          prev_layer = all_layers[(i - 1, task)]
+          if task > 0:
+            lateral_contrib, trainables = self.add_adapter(all_layers, task, i)
+            task_layers.extend(trainables)
+
+        layer = Dense(
+            in_layers=[prev_layer],
+            out_channels=layer_sizes[i],
+            activation_fn=None,
+            weights_initializer=TFWrapper(
+                tf.truncated_normal_initializer,
+                stddev=self.weight_init_stddevs[i]),
+            biases_initializer=TFWrapper(
+                tf.constant_initializer, value=self.bias_init_consts[i]))
+        task_layers.append(layer)
+
+        if i > 0 and task > 0:
+          layer = layer + lateral_contrib
+        assert self.activation_fns[i] is tf.nn.relu, "Only ReLU is supported"
+        layer = ReLU(in_layers=[layer])
+        if self.dropouts[i] > 0.0:
+          layer = Dropout(self.dropouts[i], in_layers=[layer])
+        all_layers[(i, task)] = layer
+
+      prev_layer = all_layers[(n_layers - 1, task)]
+      layer = Dense(
+          in_layers=[prev_layer],
+          out_channels=1,
+          weights_initializer=TFWrapper(
+              tf.truncated_normal_initializer,
+              stddev=self.weight_init_stddevs[-1]),
+          biases_initializer=TFWrapper(
+              tf.constant_initializer, value=self.bias_init_consts[-1]))
+      task_layers.append(layer)
+
+      if task > 0:
+        lateral_contrib, trainables = self.add_adapter(all_layers, task,
+                                                       n_layers)
+        task_layers.extend(trainables)
+        layer = layer + lateral_contrib
+      outputs.append(layer)
+      self.add_output(layer)
+      task_label = Label(shape=(None, 1))
+      task_weight = Weights(shape=(None, 1))
+      weighted_loss = ReduceSum(
+          L2Loss(in_layers=[task_label, layer, task_weight]))
+      self.create_submodel(
+          layers=task_layers, loss=weighted_loss, optimizer=None)
+    # Weight decay not activated
```

Comment:
Weight decay seems to be deactivated in the original implementation. I could not find a good way to merge in this function as well, any good idea?

Reply:
I think it should be feasible to to just add on a new loss term manually if needed. That said, I don't believe weight decay made a big difference so I'm fine to proceed without it.